Migrate from Travis-ci and Appveyor to GitHub Actions

aaronwaggener opened the pull request:
Given that travis-ci.org has now [gone read only](https://blog.travis-ci.com/2021-05-07-orgshutdown) we needed to migrate to a different provider. This is a first pass at moving to [Github Actions](https://github.com/features/actions). Let me know if you have any concerns with the approach. If this service works well for us we will migrate any other active repos that were using travis.

madelynkasula commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,35 @@
+  ELECTRON_CACHE: $HOME/.cache/electron
+jobs:
+  build-linux:
+
```
super nit: the empty- vs. no-empty-line style feels inconsistent in this file. i'm not sure if there's a convention for github actions, but [this example](https://github.com/actions/example-services/blob/main/.github/workflows/postgres-service.yml) (from the [github docs](https://docs.github.com/en/actions)) seems like a good example to follow

aaronwaggener replied:
Good catch. This was just an artifact of referencing different examples for the various items in here. I will tighten it up.

madelynkasula reviewed (commented):
i love this change! i've been interested in github actions for a while, but have yet to use it. is this something we should bring up to the rest of the eng team? i think it's okay for this to be a *labs-independent decision, but it may be something other cabals want to use or have had experience with.

also, we'll want to update the badges in the main README -- the middle badge still points to travis. not sure if this is related, but the appveyor build is failing due to a node version mismatch as well:
<img width="618" alt="Screen Shot 2021-06-24 at 9 36 31 AM" src="https://user-images.githubusercontent.com/9812299/123301193-3dcc0900-d4d0-11eb-95c0-d2eab29227ee.png">

aaronwaggener commented:
Thanks @madelynkasula. I agree, we will want to discuss this with the larger team. I spoke with Infra earlier and they are in general agreement that Actions is a good direction to go (they also do not want to mandate a solution). As long as this works well for us, I can do a PSA at an engineering weekly and/or on Slack to make sure everyone knows that their Travis workflows are no longer running and that Actions is a viable alternative.

I totally missed the badges in the Readme. Appveyor is tied to Brad's personal account. I will reach out to him to see if we can move that to our org login. I'm also not sure what value we are getting out of those builds that we can't otherwise extract from Actions. Let me know if you have any insights.

aaronwaggener commented:
@madelynkasula Appveyor was used to create builds on Windows. I have ported the job over to our Actions workflow and removed Appveyor as a dependency. I also updated the badges in our README.

madelynkasula commented:
> @madelynkasula Appveyor was used to create builds on Windows. I have ported the job over to our Actions workflow and removed Appveyor as a dependency. I also updated the badges in our README.

wonderful! one thing i noticed -- the new "Build Status" badge says "no status":
<img width="118" alt="Screen Shot 2021-06-24 at 4 56 33 PM" src="https://user-images.githubusercontent.com/9812299/123346451-2e1ee580-d50d-11eb-9cc0-d93d4c1e491a.png">

do you happen to know if this is because we haven't merged this change yet? (i.e., the "main" branch just hasn't completed a build yet?) i saw something while poking around the new actions from the "Checks" tab on this PR and found something that makes me think we might be able to configure statuses or run an action to set the status?
<img width="210" alt="Screen Shot 2021-06-24 at 4 55 27 PM" src="https://user-images.githubusercontent.com/9812299/123346531-63c3ce80-d50d-11eb-980c-7b6ef64b5bf5.png">

this could happen in a follow-up, but i figured i'd see if you already know how this piece works.

aaronwaggener commented:
> wonderful! one thing i noticed -- the new "Build" badge says "no status":

I believe that is expected as `main` has not had any commits added to it to trigger the action.

`main` branch ![Build Status](https://github.com/code-dot-org/browser/actions/workflows/main.yml/badge.svg?branch=main)

This branch ![Build Status](https://github.com/code-dot-org/browser/actions/workflows/main.yml/badge.svg?branch=GithubAction)

madelynkasula reviewed (approved):
ahhh, makes sense -- i missed the badge for this branch. this is great; i'm looking forward to using github actions!
